src/cmd/edit.rs

```diff
@@ -0,0 +1,181 @@
+use crate::cmd::{Edit, Run};
+use crate::db::{db_path, Database, DatabaseFile, Epoch, Rank};
+use crate::util::{rename, resolve_path};
+use crate::{config, util};
+use anyhow::Result;
+use std::fmt::Write as FmtWrite;
+use std::path::PathBuf;
+
+use core::mem;
+use dialoguer::{Editor, Input};
+use tempfile::tempdir;
+
+const HEADER: &str = "\
+# Blank lines and lines prepended with '#' are ignored; Line order is insignificant
+# last_accessed,rank,path
+";
+
+enum ValidationResult {
+    Success,
+    Retry,
+    Exit,
+}
+
+impl Run for Edit {
+    fn run(&self) -> Result<()> {
+        let temp_dir = tempdir()?;
+        let temp_dir_path = temp_dir.path();
+        while let Some(db_edits) = get_db_edits()? {
+            let mut db_file = DatabaseFile::new(temp_dir_path);
```

Comment:
There's no need of using a tempfile here, `db.save()` already creates a tempfile and does an atomic file rename for you. Instead, the flow could be more like:
- Open the DB.
  ```rust
  let data_dir = config::data_dir()?;
  let mut db = DatabaseFile::new(data_dir);
  let mut db = db.open()?;
  ```
- Convert it to human-readable text, allow the user to edit it, and perform any validations.
- Clear the DB and write the new directories to it.
- `db.save()?`